sdk/core/Azure.Core/src/Pipeline/Internal/RedirectPolicy.cs

```diff
@@ -0,0 +1,172 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading.Tasks;
+using Azure.Core.Diagnostics;
+
+namespace Azure.Core.Pipeline
+{
+    internal sealed class RedirectPolicy : HttpPipelinePolicy
+    {
+        private readonly int _maxAutomaticRedirections;
+
+        public static RedirectPolicy Shared { get; } = new RedirectPolicy();
+
+        private RedirectPolicy()
+        {
+            _maxAutomaticRedirections = 50;
+        }
+
+        internal async ValueTask ProcessAsync(HttpMessage message, ReadOnlyMemory<HttpPipelinePolicy> pipeline, bool async)
+        {
+            if (async)
+            {
+                await ProcessNextAsync(message, pipeline).ConfigureAwait(false);
+            }
+            else
+            {
+                ProcessNext(message, pipeline);
+            }
+
+            uint redirectCount = 0;
+            Uri? redirectUri;
+
+            Request request = message.Request;
+            Response response = message.Response;
+
+            while ((redirectUri = GetUriForRedirect(request, message.Response)) != null)
+            {
+                redirectCount++;
+
+                if (redirectCount > _maxAutomaticRedirections)
+                {
+                    // If we exceed the maximum number of redirects
+                    // then just return the 3xx response.
+                    if (AzureCoreEventSource.Singleton.IsEnabled())
+                    {
+                        AzureCoreEventSource.Singleton.RequestRedirectCountExceeded(request.ClientRequestId, request.Uri.ToString(), redirectUri.ToString());
+                    }
+
+                    break;
+                }
+
+                response.Dispose();
+
+                // Clear the authorization header.
+                request.Headers.Remove(HttpHeader.Names.Authorization);
```

Comment:
Why?

Reply:
https://github.com/dotnet/runtime/issues/26475#issuecomment-432878482

TLDR to prevent accidental secret leaks.

Reply:
Thanks.